qt/scientific_interfaces/Indirect/IndirectDataReduction.cpp

```diff
@@ -174,37 +176,39 @@ MatrixWorkspace_sptr IndirectDataReduction::instrumentWorkspace() {
  */
 void IndirectDataReduction::loadInstrumentIfNotExist(const std::string &instrumentName, const std::string &analyser,
                                                      const std::string &reflection) {
-  auto const idfDirectory = Mantid::Kernel::ConfigService::Instance().getString("instrumentDefinition.directory");
-  auto const ipfFilename = idfDirectory + instrumentName + "_" + analyser + "_" + reflection + "_Parameters.xml";
+  auto const ipfFilename = m_idfDirectory + instrumentName + "_" + analyser + "_" + reflection + "_Parameters.xml";
 
   if (ipfFilename != m_ipfFilename) {
     try {
-      auto const dateRange = instrumentName == "BASIS" ? "_2014-2018" : "";
-      auto const parameterFilename = idfDirectory + instrumentName + "_Definition" + dateRange + ".xml";
-      auto loadAlg = AlgorithmManager::Instance().create("LoadEmptyInstrument");
-      loadAlg->setChild(true);
-      loadAlg->setLogging(false);
-      loadAlg->initialize();
-      loadAlg->setProperty("Filename", parameterFilename);
-      loadAlg->setProperty("OutputWorkspace", "__IDR_Inst");
-      loadAlg->execute();
-      MatrixWorkspace_sptr instWorkspace = loadAlg->getProperty("OutputWorkspace");
+      if (!instrumentName.empty()) {
```

Comment:
It looks like nothing in this function aside from the first line is executed unless `!instrumentName.empty()`, perhaps it would be better to add this condition to the main `if` statement, and that `if` statement into an early return?

Reply:
Ah yes, that would be much better - thanks!